### DISCUSSION TITLE
Rewrite all inversions to act like "flying" inversions

### DISCUSSION THREAD

**Discussion by karst** [OpenRCT2/OpenRCT2#18424] [Feature requests]
I think it'd be a smart idea to get rid of the idea of half inversions entirely. I think it's much smarter to be able to mix and match like flying inversions can, so you can build upside down (if the track is available for it and for most with cheats ofcourse) for longer amounts of time as well (with cheats). This will get rid of quite a few lines of nearly duplicate code as well as adding the ability to make more different looking inversions. This will also allow the flying code to be the permanent code for a track type without having to have 3 separate drawing codes for essentially the same track type (straight up/flying/invert).

The inverted coaster is I believe 5 or 6 pixels taller than the flying coaster so if a coaster is listed as an invert it should just move up the drawing code by those amounts of pixels and it should be good to go. 

I imagine a project like this would be quite the undertaking but I think in the long run it will be worth putting the effort in it. 

This will also help with separating ride types and track types.

 I am not sure how all the drawing code works but I imagine that new elements would need to be added (inverted track), some removed entirely (duplicates) and some switched flying.
This would also mean that lines of code with `!trackElement.IsInverted` would need to be turned into their own separate elements rather than being a flag.

**Comment by spacek531** [comment]
When a train inverts, it pulls its sprites from a separate vehicle. Flying, Lay-Down, and Multi-Dim coaster types have two cars defined in their objects, not one - because the game switches between the two cars for uninverted and inverted positions. How would this proposed solution work for a Floorless coaster type?
